package.json

```diff
@@ -19,58 +19,74 @@
     "url": "https://github.com/mongodb-js/oidc-plugin.git"
   },
   "files": [
-    "dist"
+    "dist",
+    "index.d.ts",
+    "README.md"
   ],
   "license": "Apache-2.0",
   "main": "dist/index.js",
   "exports": {
     "require": "./dist/index.js",
     "import": "./dist/.esm-wrapper.mjs"
   },
-  "types": "./dist/index.d.ts",
+  "engines": {
+    "node": ">= 14.18.0"
+  },
+  "types": "./index.d.ts",
   "scripts": {
     "bootstrap": "npm run compile",
     "prepublishOnly": "npm run compile",
-    "compile": "tsc -p tsconfig.json && gen-esm-wrapper . ./dist/.esm-wrapper.mjs",
+    "compile": "tsc -p tsconfig.json && api-extractor run && rimraf 'dist/**/*.d.ts*' && gen-esm-wrapper . ./dist/.esm-wrapper.mjs",
     "typecheck": "tsc --noEmit",
     "eslint": "eslint",
     "prettier": "prettier",
     "lint": "npm run eslint . && npm run prettier -- --check .",
     "depcheck": "depcheck",
     "check": "npm run typecheck && npm run lint && npm run depcheck",
     "check-ci": "npm run check",
-    "test": "mocha",
+    "test": "xvfb-maybe --auto-servernum -- mocha",
+    "test-electron": "xvfb-maybe --auto-servernum -- electron-mocha",
     "test-cov": "nyc -x \"**/*.spec.*\" --reporter=lcov --reporter=text --reporter=html npm run test",
     "test-watch": "npm run test -- --watch",
     "test-ci": "npm run test-cov",
     "reformat": "npm run prettier -- --write . && npm run eslint . -- --fix"
   },
   "devDependencies": {
+    "@microsoft/api-extractor": "^7.34.4",
     "@mongodb-js/mocha-config-compass": "^0.10.0",
     "@mongodb-js/prettier-config-compass": "^0.5.0",
     "@mongodb-js/tsconfig-compass": "^0.6.0",
     "@types/chai": "^4.2.21",
     "@types/express": "^4.17.17",
     "@types/mocha": "^9.0.0",
-    "@types/node": "^17.0.35",
+    "@types/node": "^18.13.0",
+    "@types/oidc-provider": "^8.1.1",
     "@types/sinon-chai": "^3.2.5",
     "@typescript-eslint/eslint-plugin": "^5.52.0",
     "@typescript-eslint/parser": "^5.52.0",
     "chai": "^4.3.6",
     "depcheck": "^1.4.1",
+    "electron": "^23.1.0",
+    "electron-mocha": "^11.0.2",
     "eslint": "^7.25.0",
     "eslint-config-prettier": "^8.6.0",
     "eslint-plugin-mocha": "^10.1.0",
     "gen-esm-wrapper": "^1.1.0",
-    "mocha": "^8.4.0",
+    "mocha": "^10.2.0",
     "node-fetch": "^3.3.0",
     "nyc": "^15.1.0",
+    "oidc-provider": "https://github.com/panva/node-oidc-provider/archive/refs/tags/v7.14.3.tar.gz",
```

Comment:
We’re using the tarball URL here because we’re using the examples from it (and it’s not the latest version because the latest version only supports Node.js 18+), I think that’s fine for a test dependency